capstone-frontend/src/components/SearchUserModal.jsx

```diff
@@ -100,8 +130,64 @@ export class SearchUserModal extends Component {
     // Owner cannot be a Collaborator
     collaborators = collaborators.filter((collaborator) => collaborator.id !== this.props.bookList.userID);
 
-    this.setState({ addedUsersTracker: collaborators, addedUsers: collaborators });
+    const addedUsersTracker = this.removeDuplicatesArrayJsonId([...this.state.addedUsersTracker, ...collaborators]);
+
+    this.setState({ addedUsersTracker, addedUsers: collaborators });
+  }
+
+  fetchMembers = async () => {
+
+    if (!(this.props.type === 'clubs' && this.props.club)) {
+      return;
+    }
+
+    else if (this.props.club.memberIDs.length === 0) {
+      return;
+    }
+
+    let members = [];
+
+    await Promise.all(this.props.club.memberIDs.map(async (memberId) => {
+
+      const member = await fetch(`/api/user?id=${memberId}`).then(resp => resp.json());
+
+      delete member.tokenObj;
+      members.push(member);
+
+    }));
+
+    // Owner cannot be a Member
+    members = members.filter((member) => member.id !== this.props.club.ownerID);
+
+    const addedUsersTracker = this.removeDuplicatesArrayJsonId([...this.state.addedUsersTracker, ...members]);
 
+    this.setState({ addedUsersTracker, addedUsers: members });
+  }
+
+  fetchFriends = async () => {
+
+    const userID = window.localStorage.getItem('userID')
+
+    const user = await fetch(`/api/user?id=${userID}`).then(resp => resp.json());
```

Comment:
Should there be a catch here in case the fetch fails?

Reply:
Ok